Common/Source/Dialogs/dlgWayQuick.cpp

```diff
@@ -125,7 +125,35 @@ static void OnRadioFrequencyClicked(WndButton* pWnd){
 
   devPutFreqActive(Ferquency, WayPointList[SelectedWaypoint].Name);
 
-  _stprintf(szFreq,_T(" %6.3f ") ,Ferquency);
+  _stprintf(szFreq,_T("Active %6.3f ") ,Ferquency);
+
+  DoStatusMessage(_T(""), WayPointList[SelectedWaypoint].Name );
+  DoStatusMessage(_T(""), szFreq );
+  retStatus=3;
+  if(pWnd) {
+    WndForm * pForm = pWnd->GetParentWndForm();
+    if(pForm) {
+      pForm->SetModalResult(mrOK);
+    }
+  }
+#endif  // RADIO_ACTIVE
+}
+
+
+
+static void OnRadioFrequencySBClicked(WndButton* pWnd){
+#ifdef RADIO_ACTIVE
+
+  TCHAR szFreq[300];
+
+  double Ferquency;
+  LKASSERT(SelectedWaypoint>=0);
+  LKASSERT(ValidWayPointFast(SelectedWaypoint));
```

Comment:
- ValidWayPointFast() must be used only if you are inside "LockTaskData" section to avoid recursive lock
- first LKASSERT is useless, ValidWayPoint do the same...